lib/GridItem.jsx

```diff
@@ -579,29 +577,87 @@ export default class GridItem extends React.Component<Props, State> {
   ): void {
     const handler = this.props[handlerName];
     if (!handler) return;
-    const { cols, x, y, i, maxH, minH } = this.props;
+    const { cols, x, y, i, maxH, minH, useCSSTransforms } = this.props;
     let { minW, maxW } = this.props;
 
+    const handleOrientation =
+      node.classList.item(1).indexOf("w") != -1 ? "west" : "east";
     // Get new XY
     let { w, h } = calcWH(
       this.getPositionParams(),
       size.width,
       size.height,
       x,
-      y
+      y,
+      handleOrientation
     );
 
     // minW should be at least 1 (TODO propTypes validation?)
     minW = Math.max(minW, 1);
 
-    // maxW should be at most (cols - x)
-    maxW = Math.min(maxW, cols - x);
+    if (handleOrientation === "west") {
+      maxW = Math.min(maxW, cols);
+    } else {
+      maxW = Math.min(maxW, cols - x);
+    }
 
     // Min/max capping
     w = clamp(w, minW, maxW);
     h = clamp(h, minH, maxH);
 
-    this.setState({ resizing: handlerName === "onResizeStop" ? null : size });
+    const elem = ((node.parentElement: any): HTMLElement);
+    if (elem) {
+      let currentLeft, currentTop;
+      if (useCSSTransforms) {
+        const transformCSS = elem.style.transform
```

Comment:
We shouldn't have to read out the CSS, the current top/left is already in state.